monkey/monkey_island/cc/ui/src/components/report-components/zerotrust/VennDiagram/index.js

```diff
@@ -0,0 +1,369 @@
+    DOM objects.
+
+[x] I have used PropTypes already a year ago on ThreeJS/ReactJS stack project.
+
+[!] External callback is still on my list, can you make a mockup for external function which have to get pillar variable?
+
+[!] Z-indices sorting on mouseover
+    Would be n my 21.08.2019 TODO list. With D3.JS it's an easy task, however would try do 
+    make these z-soring without D3 framework.
+
+UPDATED [20.08.2019]
+
+[!] By now, there are three input props for VennDiagram component:
+    data, width and height.
+    
+[-] Since layout has to be hardcoded, it's driven by this.layout object. 
+    There're many ways of setting circles/arc positions, now I'm using the 
+    stright-forward one.
+    
+    Usually, I am put all hardcoded params to external JSON file, i.e config.json.
+    Let me know if it's a good idea.
+
+[-] Could rearange z-indecies for nodes on hover, so highlighted node would have highest z-index.
+[-] If you want callback on click even, please provide additional info what are you expecting to pass
+    through this.
+    
+[!] I don't used to make lots of comments in my code, but treying to name everything in a way, 
+    so third person could instantly get its job and concept.
+    
+    If it is not enoough just let me know.
+    
+[!] I have tried to avoid using D3 so much, especially its mouse events, cause it creates a bunch
+    of listeners for every children DOM elements. I have tried to use raw SVG objects.
+    The ArcNode is the only component where I've to call d3.arc constrictor.
+    
+[!] There are lots of discussion over different blogs an forums that ReactJS and D3.JS have a DOM
+    issue conflict, [could find lots of articels at medium.org about that, for example,
+    https://medium.com/@tibotiber/react-d3-js-balancing-performance-developer-experience-4da35f912484].
+    
+    Since the current component has only a few DOM elements, I don't thing we would have any troubles, 
+    but please keep in mind that I could tweak current version with react-faux-dom.
+    
+    Actually, by now, I'm using D3 only for math, for arc path calculations.
+    
+[!] Don't mind about code spacings, it's just for me, the final code would be clear out of them.
+
+[-] On click, an EXTERNAL callback should be called with the pillar name as a parameter. That is to enable us to expand the click functionality in future without editing the internal implementation of the component.
+
+[-] planned, [x] done, [!] see comments
+
+@author Vladimir V KUCHINOV
+@email  helloworld@vkuchinov.co.uk
+
+*/
+
+const VENN_MIN_WIDTH = 256;
+
+class ResponsiveVennDiagram extends React.Component {
+    
+  constructor(props) { super(props); }
+  render() {
+      
+    const {options, pillarsGrades} = this.props;
+    let childrenWidth = this.props.containerWidth,  childrenHeight = this.props.containerHeight;
+      
+    if(childrenHeight === 0 ||  childrenHeight === NaN){  childrenHeight = childrenWidth; } 
```

Comment:
`something === NaN` will always evaluate to `false`. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#The_necessity_of_an_isNaN_function